src/main/java/com/rakovets/course/java/core/practice/generic_types/Obscure.java

```diff
@@ -0,0 +1,43 @@
+package com.rakovets.course.java.core.practice.generic_types;
+
+public class Obscure <T> {
+    private final T object;
+
+    public Obscure(T object) {
+        this.object = object;
+    }
+
+    public T getObject() {
+        return object;
+    }
+
+    public boolean isPresent() {
+        return this.object != null;
+    }
+
+    public boolean isEmpty() {
+        return this.object == null;
+    }
+
+    public T orElse(T defaultObject) {
+        if (isPresent()) {
+            return this.getObject();
+        }
+        return defaultObject;
+    }
+
+    public T orElseThrow(Exception e) throws Exception{
```

Comment:
"Violation: There isn't separating white spaces for curly brake
Solution: Add white spaces"